src/main/java/org/folio/circulation/domain/ChangeDueDateRepository.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.circulation.domain;
+
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CompletableFuture;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ChangeDueDateRepository {
+
+  private static final Logger log = LoggerFactory
+    .getLogger(MethodHandles.lookup().lookupClass());
+
+  private final LoanRepository loanRepository;
+
+  public ChangeDueDateRepository(Clients clients) {
+    loanRepository = new LoanRepository(clients);
+  }
+
+  public CompletableFuture<Result<Loan>> changeDueDate(String loanId,
+    DateTime dueDate) {
+
+    log.info("Changing due date for {} to {}", loanId, dueDate);
```

Comment:
I thing it should be logged in `debug` level.

Reply:
Why do you think so?

Reply:
It will harm log file, every time you do a change of due date an record will be logged. 

I'm not aware what log level is used on prod but I guess it is INFO.

But maybe I'm wrong.

Reply:
I think the fact that change due date has been launched should be logged, hence INFO log level has been chose.n

Reply:
ok, no prob, let's leave it in INFO

Reply:
Just an FYI, we've found excessive logging is slowing down services, like Okapi. While I don't know if it is the case for mod-circulation, we'll likely find out over the next few weeks.

Either way, logging isn't free and I think we'll need to seriously consider the impact of logging, the various logging implementations we use, what we log, how/when we generate the message strings and at what level at some point in the near future.

Also, the amount of existing INFO level logging makes mod-circulation fairly difficult to manually digest. A single checkin writes thousands of lines of INFO messages.

Note, I am ***not*** suggesting changing the level of this message for this PR. This is a topic for another time.

Reply:
@mreno-EBSCO I agree, this likely needs significant review (and should probably be raised as technical debt).

Are many of the existing info messages requests made to other APIs and responses? If so, they should likely be lowered to debug only.

I imagine we need a way of changing the logging level. Defaulting to warnings and errors only.

If we review which logging framework to use, I'd like to use one that allows for deferred execution of the message, so we can avoid checking log levels directly.

Reply:
@marcjohnson-kint yes, most of the INFO messages are the JSON bodies of request/responses to other modules.